scene/animation/tween.cpp

```diff
@@ -713,6 +713,7 @@ void Tween::_tween_process(float p_delta) {
 
 		// Is the data not active or already finished? No need to go any further
 		if (!data.active || data.finish) {
+			all_finished = true;
```

Comment:
The fix would be to fix the logic for `all_finished` on line 716, not hack it here, possibly losing information about a previous `data` which was active and not finished (so this would introduce a bug IMO).

Reply:
Is it better to make this ?
`all_finished = !data.active || data.finish;`

Reply:
No, that would erase the value from previous iterations.
It should be (on line 712):
```
all_finished = all_finished && (data.finish || !data.active)
```

Reply:
but `all_finished` is false at this moment and it's because it false that `set_active(false)` and `emit_signal("tween_all_completed")` was not sent (line 816)

Reply:
In your test project, sure. But if could be true if the previous iteration was true.
Its purpose is to check that *all* iterations are finished (or, as you're trying to change, inactive). If *any* iteration is active and not finished, it should end up false when the loop terminates.

Reply:
After the last amend we're back to the initial state, which is wrong. See https://github.com/godotengine/godot/pull/39801#discussion_r457329250